src/approves/is-pr-fully-approved.ts

```diff
@@ -1,18 +1,34 @@
-/*
-Function that returns true if PR is fully approved.
-PR is considered NOT fully approved when:
-  - some changes requested at current state
-  - some changes requested that was not approved lately
-  - required PR checks are failed.
+import { Rule, Reviews, Checks } from '../config/typings';
+import { warning } from '../logger';
+import { checkReviewersRequiredChanges } from './identify-reviews';
+import { areCIChecksPassed } from './identify-ci';
 
-If PR is NOT fully approved,
-  return a reason why:
-    - groups that still waiting for an approval (how many left)
-    - users that have requested changes (now or before and it was not approved lately)
-    - failed CI checks
+type Params = {
+  rules: Rule[];
+  requiredChecks: string[] | undefined;
+  checks: Checks;
+  reviews: Reviews;
+};
 
-Arguments are:
-  - currentState (result of identifyCurrentState function)
-  - approver groups (result of identifyApprovers function)
-  - CI checks statuses @TODO
-*/
+export function isPrFullyApproved({
+  rules,
+  requiredChecks,
+  checks,
+  reviews,
+}: Params): boolean {
+  const checkCIChecks = areCIChecksPassed(checks, requiredChecks);
+
+  if (checkCIChecks !== true) {
+    warning(checkCIChecks);
```

Comment:
Let's change logs here to `info`. This is normal behaviour of our program.
On top of that, I'd return error string here or true: 
- error explaining why it's not ready to be merged
- true if all ok